core/src/test/java/org/elasticsearch/index/mapper/date/SimpleDateMappingTests.java

```diff
@@ -289,7 +288,7 @@ public void testIgnoreMalformedOption() throws Exception {
                     .endObject()
                     .bytes());
         } catch (MapperParsingException e) {
-            assertThat(e.getCause(), instanceOf(MapperParsingException.class));
+            assertThat(e.getCause(), instanceOf(IllegalArgumentException.class));
```

Comment:
can we check part of the message so we know we got the right exception? ditto for the other cases like this in this file